app/controllers/SurveyController.java

```diff
@@ -190,7 +190,7 @@ public Result createSurvey() throws Exception {
         UserSession session = getAuthenticatedResearcherSession();
         StudyIdentifier studyId = session.getStudyIdentifier();
         
-        Survey survey = DynamoSurvey.fromJson(requestToJSON(request()));
+        Survey survey = parseJson(request(), Survey.class);
         survey.setStudyIdentifier(studyId.getIdentifier());
```

Comment:
(Never mind earlier comment. It looks like you're preserving that factory method for survey elements.)